hw/ip/rom_ctrl/doc/dv/index.md

```diff
@@ -0,0 +1,102 @@
+In addition, it instantiates the following interfaces, connects them to the DUT and sets their handle into `uvm_config_db`:
+* [Clock and reset interface]({{< relref "hw/dv/sv/common_ifs" >}})
+* [TileLink host interface]({{< relref "hw/dv/sv/tl_agent/README.md" >}})
+* [Memory backdoor interface]({{< relref "hw/dv/sv/mem_bkdr_if/README.md" >}})
+* ROM_CTRL IOs
+* Alerts ([`pins_if`]({{< relref "hw/dv/sv/common_ifs" >}})
+
+### Common DV utility components
+The following utilities provide generic helper tasks and functions to perform activities that are common across the project:
+* [dv_utils_pkg]({{< relref "hw/dv/sv/dv_utils/README.md" >}})
+* [csr_utils_pkg]({{< relref "hw/dv/sv/csr_utils/README.md" >}})
+
+### Compile-time configurations
+There is only one compile-time configuration, where arbitrary values are chosen for compile-time constants.
+
+### Global types & methods
+All common types and methods defined at the package level can be found in
+`rom_ctrl_env_pkg`. Some of them in use are:
+```systemverilog
+  // arbitrary compile-time constants
+  parameter bit [63:0]  RNDCNSTSCRNONCE = 64'h0123456789ABCDEF;
+  parameter bit [127:0] RNDCNSTSCRKEY   = 128'hFEDCBA9876543210FEDCBA9876543210;
+```
+### TL_agent
+ROM_CTRL testbench instantiates (already handled in CIP base env) [tl_agent]({{< relref "hw/dv/sv/tl_agent/README.md" >}})
+which provides the ability to drive and independently monitor random traffic via both
+TL host interfaces into ROM_CTRL device.
+
+### UVM RAL Model
+The ROM_CTRL RAL model is created with the [`ralgen`]({{< relref "hw/dv/tools/ralgen/README.md" >}}) FuseSoC generator script automatically when the simulation is at the build stage.
+
+It can be created manually by invoking [`regtool`]({{< relref "util/reggen/README.md" >}}):
+
+### Stimulus strategy
+#### Test sequences
+All test sequences reside in `hw/ip/rom_ctrl/dv/env/seq_lib`.
+The `rom_ctrl_base_vseq` virtual sequence is extended from `cip_base_vseq` and serves as a starting point.
+All test sequences are extended from `rom_ctrl_base_vseq`.
+It provides commonly used handles, variables, functions and tasks that the test sequences can simple use / call.
+Some of the most commonly used tasks / functions are as follows:
+* rom_ctrl_mem_init: initialize the rom to random values
+* do_rand_ops: drive random traffic into the rom TLUL interface
+
+#### Functional coverage
+To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
+The following covergroups have been developed to prove that the test intent has been adequately met:
+* TODO
+
+### Self-checking strategy
+#### Scoreboard
+The `rom_ctrl_scoreboard` is primarily used for end to end checking.
+It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
+* kmac_req_fifo
+* kmac_resp_fifo
+
+The scoreboard monitors traffic sent to and from the kmac interface.
```

Comment:
Nit: I'd consider either writing "KMAC" or `kmac`. The upper-casing or font change makes it a bit more obvious this is a code-like thing. Similarly for "rom TLUL" below.